assets/js/googlesitekit/datastore/user/user-input-settings.js

```diff
@@ -325,6 +349,43 @@ const baseSelectors = {
 			return settings[ settingID ]?.author;
 		}
 	),
+
+	/**
+	 * Indicates whether the current user input settings have changed from what is saved.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object}     state Data store's state.
+	 * @param {Array|null} keys  Settings keys to check; if not provided, all settings are checked.
+	 * @return {boolean} True if the settings have changed, false otherwise.
+	 */
+	haveUserInputSettingsChanged( state, keys = null ) {
+		const { inputSettings, savedInputSettings } = state;
+
+		if ( keys ) {
+			return ! isEqual(
+				pick( inputSettings, keys ),
+				pick( savedInputSettings, keys )
+			);
+		}
+
+		return ! isEqual( inputSettings, savedInputSettings );
+	},
+
+	/**
+	 * Indicates whether the provided user input setting has changed from what is saved.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} state   Data store's state.
+	 * @param {string} setting The setting we want to check for saved changes.
+	 * @return {boolean} True if the settings have changed, false otherwise.
+	 */
+	hasUserInputSettingChanged( state, setting ) {
+		invariant( setting, 'setting is required.' );
+
+		return baseSelectors.haveUserInputSettingsChanged( state, [ setting ] );
```

Comment:
Passing `setting` within an array here because `haveUserInputSettingsChanged` expects the `keys` param to be an array.